pkg/tools/streamutils/streamutils.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package streamutils provides utils for registry streams
+package streamutils
+
+import "github.com/networkservicemesh/api/pkg/api/registry"
+
+// ReadNetworkServiceList read list of NetworkServices from passed stream
+func ReadNetworkServiceList(stream registry.NetworkServiceRegistry_FindClient) []*registry.NetworkService {
```

Comment:
LOL... its times like these when I wish Go 2.0 Generics had landed :)

I'm fine merging these here now, but would they perhaps not make more sense moved up to the api/ repo as helper functions?

Reply:
@edwarnicke sounds smart, I think we need to move it to api

Reply:
Yeah... let's get this in first.

Reply:
Thanks, done!